### PR TITLE
Add token auth

### PR DESCRIPTION
**Pull request by dchiquit**

### CODE REVIEW

File: publish/api.py [hunk 1]
```diff
@@ -19,6 +20,13 @@ class DandisetViewSet(viewsets.ModelViewSet):
     serializer_class = DandisetSerializer
     pagination_class = DandisetPagination
 
+    def get_permissions(self):
+        if self.action in ['retrieve', 'list']:
+            permission_classes = []
+        else:
+            permission_classes = [IsAuthenticated]
+        return [permission() for permission in permission_classes]
+
```
Comment (jbeezley) [inline publish/api.py:29]: Could you make this a permission class to deduplicate the function?

**Comment by jbeezley** [comment]
LGTM

File: publish/api.py [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
 from .models import Dandiset, NWBFile, Subject
 from .serializers import DandisetSerializer, DandisetPublishSerializer, NWBFileSerializer, SubjectSerializer
 from .tasks import publish_dandiset
 
 
+class PermissionMixin:
```
Comment (brianhelba) [inline publish/api.py:12]: Why not use [`IsAuthenticatedOrReadOnly`](https://www.django-rest-framework.org/api-guide/permissions/#isauthenticatedorreadonly)?

Comment (dchiquit) [reply publish/api.py:12]: I had not heard of it. I will include that in my next PR

**Pull request merged**